heatmap_printer.go

```diff
@@ -0,0 +1,807 @@
+package pterm
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"math"
+	"strings"
+
+	"github.com/pterm/pterm/internal"
+)
+
+// DefaultHeatmap contains standards, which can be used to print a HeatmapPrinter.
+var DefaultHeatmap = HeatmapPrinter{
+	AxisStyle:                  &ThemeDefault.HeatmapHeaderStyle,
+	SeparatorStyle:             &ThemeDefault.HeatmapSeparatorStyle,
+	VerticalSeparator:          "│",
+	TopRightCornerSeparator:    "└",
+	TopLeftCornerSeparator:     "┘",
+	BottomLeftCornerSeparator:  "┐",
+	BottomRightCornerSeparator: "┌",
+	HorizontalSeparator:        "─",
+	TSeparator:                 "┬",
+	TReverseSeparator:          "┴",
+	LSeparator:                 "├",
+	LReverseSeparator:          "┤",
+	TCrossSeparator:            "┼",
+	LegendTag:                  "Legend",
+	Boxed:                      true,
+	Grid:                       true,
+	Legend:                     true,
+	TextRGB:                    RGB{0, 0, 0, false},
+	RGBRange:                   []RGB{{R: 255, G: 0, B: 0, Background: true}, {R: 255, G: 165, B: 0, Background: true}, {R: 0, G: 255, B: 0, Background: true}},
+	TextColor:                  FgBlack,
+	Colors:                     []Color{BgRed, BgLightRed, BgYellow, BgLightYellow, BgLightGreen, BgGreen},
+
+	EnableRGB: false,
+
+	rgbLegendValue: 11,
+}
+
+// HeatmapData is the type that contains the data of a HeatmapPrinter.
+type HeatmapData [][]float32
+
+type HeatmapAxis struct {
+	XAxis []string
+	YAxis []string
+}
+
+// HeatmapPrinter is able to render tables.
+type HeatmapPrinter struct {
+	HasHeader                  bool
+	AxisStyle                  *Style
+	VerticalSeparator          string
+	TopRightCornerSeparator    string
+	TopLeftCornerSeparator     string
+	BottomLeftCornerSeparator  string
+	BottomRightCornerSeparator string
+	HorizontalSeparator        string
+	TSeparator                 string
+	TReverseSeparator          string
+	LSeparator                 string
+	LReverseSeparator          string
+	TCrossSeparator            string
+	LegendTag                  string
+	SeparatorStyle             *Style
+	Data                       HeatmapData
+	Axis                       HeatmapAxis
+	Boxed                      bool
+	Grid                       bool
+	OnlyColoredCells           bool
+	LegendOnlyColoredCells     bool
+	ColorValueComplementarily  bool
+	Legend                     bool
+	CellSize                   int
+	Colors                     []Color
+	TextColor                  Color
+	EnableRGB                  bool
+	RGBRange                   []RGB
+	TextRGB                    RGB
+	Writer                     io.Writer
+
+	minValue float32
+	maxValue float32
+
+	rgbLegendValue int
+}
+
+var colorComplement = map[Color]Color{
```

Comment:
Maybe rename to `complementaryColors`?

Reply:
we talked about this a few months and decided on this naming